pkg/aspect/applicationLogsManager.go

```diff
@@ -17,56 +17,49 @@ package aspect
 import (
 	"encoding/json"
 	"fmt"
+	"text/template"
 	"time"
 
+	"github.com/golang/glog"
+	"github.com/hashicorp/go-multierror"
+
 	dpb "istio.io/api/mixer/v1/config/descriptor"
 	"istio.io/mixer/pkg/adapter"
 	aconfig "istio.io/mixer/pkg/aspect/config"
 	"istio.io/mixer/pkg/attribute"
 	"istio.io/mixer/pkg/config"
 	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/pool"
 	"istio.io/mixer/pkg/status"
 )
 
 type (
+	// PayloadFormat describes the format of the LogEntry's payload.
+	PayloadFormat int
+
 	applicationLogsManager struct{}
 
+	logInfo struct {
+		format    PayloadFormat
+		severity  string
+		timestamp string
+		tmpl      *template.Template
+		tmplExprs map[string]string
+		labels    map[string]string
+	}
+
 	applicationLogsWrapper struct {
-		logName            string
-		descriptors        []dpb.LogEntryDescriptor // describe entries to gen
-		inputs             map[string]string        // map from param to expr
-		severityAttribute  string
-		timestampAttribute string
-		timestampFmt       string
-		aspect             adapter.ApplicationLogsAspect
-		defaultTimeFn      func() time.Time
+		name     string
+		aspect   adapter.ApplicationLogsAspect
+		metadata map[string]*logInfo // descriptor_name -> info
 	}
 )
 
-var (
-	defaultLog = dpb.LogEntryDescriptor{
-		Name:             "default",
-		DisplayName:      "Default Log Entry",
-		Description:      "Placeholder log descriptor",
-		PayloadAttribute: "logMessage",
-		Attributes: []string{
-			"serviceName",
-			"peerId",
-			"operationId",
-			"operationName",
-			"apiKey",
-			"url",
-			"location",
-			"apiName",
-			"apiVersion",
-			"apiMethod",
-			"requestSize",
-			"responseSize",
-			"responseTime",
-			"originIp",
-			"originHost",
-		},
-	}
+const (
+	// TEXT describes a log entry whose TextPayload should be set.
+	TEXT = iota
```

Comment:
should we just have this be `Text`? I don't think we need all caps (not an initialism, like JSON).

Also, should these be PayloadFormats? I think we want that typing reflected here.